chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -440,10 +440,15 @@ class Bundle extends Record {
   /** Returns a field's contained user-defined Bundle element if it appears to
     * be one, otherwise returns None.
     */
-  private def getBundleField(m: java.lang.reflect.Method): Option[Data] = m.invoke(this) match {
-    case d: Data => Some(d)
-    case Some(d: Data) => Some(d)
-    case _ => None
+  private def getBundleField(m: java.lang.reflect.Method): Option[Data] = {
+    if (m.getDeclaringClass.isAssignableFrom(classOf[Bundle])) None
```

Comment:
So I just got to trying what you suggested and no tests pass (compiles OK, but runtime derps):
```
[info]   chisel3.core.Binding$BindingException: 'this' (chisel3.core.UInt@7): Not bound to synthesizable node, currently only Type description
[info]   at chisel3.core.Binding$.checkSynthesizable(Binding.scala:186)
[info]   at chisel3.core.Bits.do_apply(Bits.scala:100)
[info]   at chisel3.core.Bits.do_apply(Bits.scala:112)
[info]   at chiselTests.CompatibiltySpec$$anonfun$13$RequireIOWrapModule$1.<init>(CompatibiltySpec.scala:206)
[info]   at chiselTests.CompatibiltySpec$$anonfun$13$$anonfun$apply$mcV$sp$8.apply(CompatibiltySpec.scala:208)
[info]   at chiselTests.CompatibiltySpec$$anonfun$13$$anonfun$apply$mcV$sp$8.apply(CompatibiltySpec.scala:208)
[info]   at chisel3.core.Module$.do_apply(Module.scala:42)
[info]   at chisel3.Driver$$anonfun$elaborate$1.apply(Driver.scala:91)
```

```
info] - Bundles with same data but different, underlying elements should compare as UInt *** FAILED ***
[info]   chisel3.core.Binding$BindingException: 'this' (chisel3.core.UInt@a): Not bound to synthesizable node, currently only Type description
[info]   at chisel3.core.Binding$.checkSynthesizable(Binding.scala:186)
[info]   at chisel3.core.Data.connect(Data.scala:186)
[info]   at chisel3.core.Data.$colon$eq(Data.scala:258)
[info]   at chiselTests.BundleToUnitTester.<init>(BundleWire.scala:38)
[info]   at chiselTests.BundleToUIntSpec$$anonfun$3$$anonfun$apply$mcV$sp$2.apply(BundleWire.scala:70)
[info]   at chiselTests.BundleToUIntSpec$$anonfun$3$$anonfun$apply$mcV$sp$2.apply(BundleWire.scala:70)
```

Pushed changes to https://github.com/ucb-bar/chisel3/commits/builderreflectionfix if you want to take a look. If I'm not doing something stupid (copy pasting incorrectly), then maybe it's better to just consider merging this PR...